sktime/transformations/compose.py

```diff
@@ -663,3 +664,137 @@ def _underscore_join(iterable):
         """Create flattened column names from multiindex tuple."""
         iterable_as_str = [str(x) for x in iterable]
         return "__".join(iterable_as_str)
+
+
+class FitInTransform(BaseTransformer):
+    """Transformer compostion to always fit a given transformer on the transform data only.
+
+    Some transformers like Imputer(method="mean") might be more accurate for some use
+    cases if they are fitted only on the transform/predict data. The FitInTransform
+    transformer can be best used in a pipeline context where train data is different to
+    the transform/predict data.
+
+    Parameters
+    ----------
+    transformer : Estimator
+        scikit-learn-like or sktime-like transformer to fit and apply to series.
+    skip_inverse_transform : bool
+        The FitInTransform will skip inverse_transform by default, of the param
+        skip_inverse_transform=False, then the inverse_transform is calculated
+        by means of transformer.fit(X=X, y=y).inverse_transform(X=X, y=y) where
+        transformer is the inner transformer. So the inner transformer is
+        fitted on the inverse_transform data. This is required to have a non-
+        state changing transform() method of FitInTransform.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_longley
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.compose import ForecastingPipeline
+    >>> from sktime.forecasting.model_selection import temporal_train_test_split
+    >>> from sktime.transformations.compose import FitInTransform
+    >>> from sktime.transformations.series.impute import Imputer
+
+    >>> y, X = load_longley()
+    >>> y_train, y_test, X_train, X_test = temporal_train_test_split(y, X)
+    >>> fh = ForecastingHorizon(y_test.index, is_relative=False)
+
+    >>> # we want to fit the Imputer only on the predict (=transform) data.
+    >>> # note that NaiveForecaster cant use X data, this is just a show case.
+    >>> pipe = ForecastingPipeline(
```

Comment:
I´d recommend a `ClassifierPipeline` becaues there the algorithm actually makes sense...

Reply:
sure we can add this as well, can you make a proposal? I am not so familiar with the classification module

Reply:
`pipe = imputer * SummaryClassifier()`

Reply:
I think we shouldnt use dunder methods in examples, if s.o. sees it first time it just looks confusing

Reply:
so just pick one of the examples from `ClassificationPipeline` and replace the transformer

Reply:
i dont find a class `ClassificationPipeline`?